x/posts/internal/keeper/querier.go

```diff
@@ -90,3 +100,39 @@ func queryPosts(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) ([]byte,
 
 	return bz, nil
 }
+
+//queryPollAnswers handles the request to get poll answers related to a post with given id
+func queryPollAnswers(ctx sdk.Context, path []string, _ abci.RequestQuery, keeper Keeper) ([]byte, error) {
+	id, err := types.ParsePostID(path[0])
+
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, fmt.Sprintf("Invalid post id: %s", path[0]))
+	}
+
+	post, found := keeper.GetPost(ctx, id)
+	if !found {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, fmt.Sprintf("Post with id %s not found", id))
+	}
+
+	if post.PollData == nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Post with id %s has no poll associated", id))
+	}
+
+	pollAnswers := keeper.GetPollAnswers(ctx, id)
+
+	if pollAnswers == nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Post with id %s has no answers to poll", id))
```

Comment:
Why are we throwing an error if a poll does not yet have any answer to it? I think we should simply return an empty slice instead.